designer/fitting/dwipi.py

```diff
@@ -65,6 +65,26 @@ def __init__(self, imPath):
             assert('File in path not found. Please locate file and try '
                    'again')
         tqdm.write('Image ' + fName + '.nii loaded successfully')
+        if not isinstance(nthreads, int):
+            assert('Variable nthreads need to be an integer')
+        if nthreads < -2:
+            assert('Variable nthreads cannot be set below -2')
+        if nthreads is None:
+            self.workers = -1
+        else:
+            self.workers = nthreads
+        if self.workers < -1:
+            tqdm.write('Processing with ' +
+                       np.str(multiprocessing.cpu_count() + self.workers + 1)
```

Comment:
Why add self.workers and 1?

Reply:
That's the formula for number of workers when `nthreads < -1`

Reply:
That's confusing. Just disallow values <= -2 and values of 0. Then make the logic for -1
`self.workers = multiprocessing.cpu_count()`